config/tasks/cloud_agents.yml

```diff
@@ -2,6 +2,37 @@
 
 - name: cloud_agents | Qualys cloud agent
   block:
+    - name: cloud_agents | Retrieve Qualys agents from github
+      uri:
+        url: "{{ cloud_agent.qualys.github_release }}"
+        force_basic_auth: yes
+        url_username: "{{ environment_credentials.github_username }}"
+        url_password: "{{ environment_credentials.github_password }}"
+        headers:
+          Accept: "application/vnd.github.v3+json"
+      register: github_list_qualys
+    - set_fact: 
+        qualys_url: "{{ github_list_qualys | json_query('json[].assets[?name==`'+ cloud_agent.qualys.bin_name +'`].url[] | [0]') }}"
+
+    - name: cloud_agents | Get latest Qualys agent
+      uri:
+        url: "{{ qualys_url }}"
```

Comment:
Instead of creating the `qualys_url` fact, you could just set the url directly:
`url: "{{ github_list_qualys | json_query('json[].assets[?name==`'+ cloud_agent.qualys.bin_name +'`].url[] | [0]') }}"`

register variables are already global (unfortunately), so this saves adding two variables instead of one to the global vars (unless you need to use it globally?).

Reply:
I have made the changes